### PR TITLE
doc: release-notes: Summarize Bluetooth Mesh changes

### PR DESCRIPTION
**Pull request by trond-snekvik**
Includes upstream and downstream changelog since 1.4.

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

### CODE REVIEW

**Review by ru-fu — CHANGES_REQUESTED** [review]
Will add a commit with changes.

**Review by ru-fu — COMMENTED** [review]
Added a commit - please check and squash if okay.

File: doc/nrf/releases/release-notes-latest.rst [hunk 1]
```diff
@@ -107,3 +115,17 @@ The current |NCS| release is based on Zephyr 2.4.0.
...
+* Bluetooth Mesh:
+
+  * Replaced the Configuration Server structure with Kconfig entries and a standalone Heartbeat API.
+  * Added a separate API for adding keys and configuring features locally.
+  * Fixed a potential infinite loop in model extension tree walk.
+  * Added LPN and Friendship event handler callbacks.
+  * Created separate internal submodules for keys, labels, Heartbeat, replay protection, and feature management.
+  * :ref:`bluetooth_mesh_models_cfg_cli`:
+
+    * Added an API for resetting a node (:c:func:`bt_mesh_cfg_node_reset`).
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:130]: Are these the correct functions?

Comment (trond-snekvik) [reply doc/nrf/releases/release-notes-latest.rst:130]: Yes, thanks!

**Comment by trond-snekvik** [comment]
Rebased and squashed.

**Review by ru-fu — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Pull request merged**